### PR TITLE
WIP: Add, edit, delete Alerts

### PR DESCRIPTION
**Pull request by scotthallock**
## Overview

User can now add, edit, and delete alerts.
- Removed Mock data
- REST API on extensionServer (will create controllers on future PR)
- Popup alerts to provide error feedback to users (example: invalid email)
- User's alerts are stored in an array on the extensionServer

Known issues for future PR:
- Required fields in form are not being enforced
- Need to connect the alerts to the metrics monitoring

### Screenshots
Screenshot 2023-07-03 at 9 58 44 PM
Screenshot 2023-07-03 at 9 58 54 PM
Screenshot 2023-07-03 at 9 59 02 PM

### CODE REVIEW

File: backend/package.json [hunk 1]
```diff
@@ -11,11 +11,14 @@
   "license": "ISC",
   "dependencies": {
     "axios": "^1.4.0",
+    "email-validator": "^2.0.4",
```
Comment (scotthallock) [inline backend/package.json:14]: for validating user emails on a future feature

File: backend/src/extensionServer.ts [hunk 2]
```diff
@@ -24,4 +29,87 @@ app.get(
   }
 );
 
-export default app;
\ No newline at end of file
+// =============== Request handlers for Alerts functionality ===============
```
Comment (scotthallock) [inline backend/src/extensionServer.ts:32]: Will create controllers for these request handlers on a future PR

File: ui/src/components/AlertDialog/AlertDialog.tsx [hunk 3]
```diff
@@ -16,30 +15,42 @@ import {
...
 
 const metricHelperText = {
   'CPU %': '% of CPU used by the container based on the number of cores allocated',
   'MEM %': '% of memory used by the container',
 };
 
-export default function AlertDialog({ ddClient, dialogOpen, setDialogOpen }: AlertDialogProps) {
+const dialogTitle = {
+  NEW: 'Create Alert',
+  EDIT: 'Edit Alert',
```
Comment (patrickv77) [inline ui/src/components/AlertDialog/AlertDialog.tsx:27]: Is there a reason for the keys to be all uppercase here?

**Review by patrickv77 — APPROVED** [review]
Awesome work! As usual, the UI is clean and intuitive. Love the notes under each input box.

File: ui/src/pages/Alerts/Alerts.tsx [hunk 4]
```diff
@@ -88,10 +194,17 @@ function AlertCard({
           }}
         >
           <Typography variant="h3">{name}</Typography>
-          <IconButton sx={{ ml: 'auto' }}>
+          <IconButton
+            sx={{ ml: 'auto' }}
+            onClick={() => {
+              // In the EDIT mode, populate the dialog form fields with all the alert settings
```
Comment (patrickv77) [inline ui/src/pages/Alerts/Alerts.tsx:200]: Really awesome attention to detail!

File: ui/src/components/AlertDialog/AlertDialog.tsx (hunk 3 above)
Comment (scotthallock) [reply ui/src/components/AlertDialog/AlertDialog.tsx:27]: The values on this object are accessed using `dialogSettings.mode` which is an alias for `<"NEW" | "EDIT">`

**Pull request merged**